### PR TITLE
ESC docs improvements

### PR DESCRIPTION
**Pull request by interurban**
WIP improvements to create better sections for ESC docs and centralize content.

Todo
- [x] Adjust all the blog backlinks
- [x] edit esc-cli.yml for auto-gen cli docs
- [ ] cross link which issues will close out on https://github.com/pulumi/docs/issues/12594

### CODE REVIEW

File: scripts/redirects/esc-cli-redirects.txt
Comment (cnunciato) [inline scripts/redirects/esc-cli-redirects.txt:1]: I see 17 redirects here, but more than 17 changes in the actual files. Looks like some didn't make it onto the list?

File: content/docs/esc/environments/dynamic-environment-variables.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+menu:
+  pulumiesc:
+    identifier: esc-dynamic-environmeant-variables
+    parent: esc-environments
+    weight: 4
+---
+
+The Pulumi ESC CLI includes a [`run` command](/docs/esc-cli/commands/esc_run/) that allows you to run commands with Pulumi ESC managed environment variables, without exporting them to your shell.
+
+## Running commands with environment variables
```
Comment (cnunciato) [inline content/docs/esc/environments/dynamic-environment-variables.md:15]: I don't think we need this heading here and suggest we use it for the H1 instead.

File: content/docs/esc/environments/versioning.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+```bash
+$ esc env diff myorg/test@3 myorg/test@2
+ Value
+
+    --- myorg/test@3
+    +++ myorg/test@2
+...
+```
+
+### Tagging versions
```
Comment (cnunciato) [inline content/docs/esc/environments/versioning.md:43]: Ah, I see -- though this should probably be an H2:

```suggestion
## Tagging versions
```

File: content/docs/esc/_index.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ aliases:
 
 Pulumi ESC (Environments, Secrets, and Configuration) allows teams to tackle secrets and configuration complexity for modern cloud environments, alleviating maintenance burden and reducing costly mistakes, and creating a "secure by default" posture. Pulumi ESC is a new category of configuration as code product, motivated by our experience working with hundreds of Pulumi IaC customers to address their needs in managing secrets and configuration at scale within their Pulumi infrastructure and across other cloud applications and infrastructure projects.
```
Comment (aaronkao) [inline content/docs/esc/_index.md:16]: should the URL instead be "secrets-management"?

File: content/docs/esc/environments/working-with-environments.md [hunk 4]
```diff
@@ -760,4 +501,6 @@ Please confirm that this is what you'd like to do by typing `myorg/test`: myorg/
 Environment "myorg/test" has been removed!
 ```
 
+## Renaming an environment
+
 Environments cannot be renamed.
```
Comment (cnunciato) [inline content/docs/esc/environments/working-with-environments.md:506]: I like the addition of the heading, but it's a bit of a bummer to see it, click it, and then be told what it says isn't possible. How about suggesting they create a copy of the environment and delete the old one? Same end result and wouldn't feel as much like a dead end.

File: content/docs/esc/integrations/_index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+h1: Pulumi ESC integrations
+menu:
+    pulumiesc:
+        weight: 8
+        identifier: esc-integrations
+aliases:
+    - /docs/esc/providers/
+---
+
+Pulumi ESC support for dynamic configuration providers allow ESC to integrate with secrets stored in any other provider. Organizations often use AWS OIDC, AWS Secrets Manager, Vault, Azure OIDC, Azure KeyVault, GCP OIDC, and GCP Secrets Manager plus many more sources of truth for their secrets and configuration. Pulumi ESC supports them all, providing a single interface to your configuration and secrets, no matter where their source of truth is. Pulumi ESC works with these tools to provide improved management of secrets and configuration.
```
Comment (cnunciato) [inline content/docs/esc/integrations/_index.md:14]: support ... allows:

```suggestion
Pulumi ESC's support for dynamic configuration providers allows ESC to integrate with secrets stored in any other provider. Organizations often use AWS OIDC, AWS Secrets Manager, Vault, Azure OIDC, Azure KeyVault, GCP OIDC, and GCP Secrets Manager plus many more sources of truth for their secrets and configuration. Pulumi ESC supports them all, providing a single interface to your configuration and secrets, no matter where their source of truth is. Pulumi ESC works with these tools to provide improved management of secrets and configuration.
```

File: content/docs/esc/integrations/_index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ESC also integrates with tools like Direnv, Terraform, and Docker to help manage environment variables, interact with existing infrastructure, and support containerized workflows.
+
+## Dynamic login providers
+
+- [AWS login provider](/docs/esc/integrations/dynamic-login-credentials/aws-login)
+- [Azure login provider](/docs/esc/integrations/dynamic-login-credentials/azure-login)
+- [GCP login provider](/docs/esc/integrations/dynamic-login-credentials/gcp-login)
+- [Vault login provider](/docs/esc/integrations/dynamic-login-credentials/vault-login)
+
+## Dynamic secrets providers
```
Comment (cnunciato) [inline content/docs/esc/integrations/_index.md:25]: As an editorial note, I feel like we're leaning too heavily on the word "dynamic" in a lot of this stuff, and while I get it, it does feel a bit vague and repetitive. The key thing we want people to see is that there are two kinds of integration -- one for obtaining provider credentials specifically, the other for obtaining third-party secrets more generally. The "dynamic"-ness of both is sort of a given (i.e., in that we "fetch stuff") and seems superfluous, considering there isn't really a "static" version of either one, if that makes sense.

File: content/docs/esc/integrations/dynamic-secrets/_index.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+---
+title: Dynamic Secrets
```
Comment (cnunciato) [inline content/docs/esc/integrations/dynamic-secrets/_index.md:2]: Yeah, again -- it's really not about "dynamic secrets", it's about "fetching and using secrets from  other providers".

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+h1: ESC Pulumi IaC Integration
+meta_desc: Pulumi ESC integrates with Pulumi IaC to expose environment settings and secrets to Pulumi stacks, simplifying configuration management.
+menu:
+  pulumiesc:
+    identifier: esc-pulumi-iac-integrations
+    parent: esc-inf-tools-integrations
+    weight: 1
+---
+
+## Using environments with Pulumi IaC
```
Comment (cnunciato) [inline content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:14]: I don't think we need a separate heading here. We should just make this the title/H1 of the page.

File: content/docs/esc/integrations/infrastructure/cloudflare.md [hunk 9]
```diff
@@ -3,11 +3,11 @@ title_tag: Integrate with Cloudflare | Pulumi ESC
...
 h1: "Pulumi ESC: Integrate with Cloudflare"
 meta_desc: This page provides an overview of how to use Pulumi ESC with Cloudflare.
-weight: 2
 menu:
   pulumiesc:
-    parent: esc-other-integrations
-    identifier: esc-other-integrations-cloudflare
+    parent: esc-inf-tools-integrations
+    identifier: esc-inf-tools-integrations-cloudflare
+    weight: 4
```
Comment (cnunciato) [inline content/docs/esc/integrations/infrastructure/cloudflare.md:10]: Missing an alias.

File: content/docs/esc/integrations/infrastructure/pulumi-iac/pulumi-stacks.md [hunk 10]
```diff
@@ -1,16 +1,15 @@
...
 menu:
     pulumiesc:
         identifier: pulumi-stacks
-        parent: esc-providers
-        weight: 7
+        parent: esc-pulumi-iac-integrations
+        weight: 1
 aliases:
-- /docs/pulumi-cloud/esc/providers/pulumi-stacks/
+     - /docs/pulumi-cloud/esc/providers/pulumi-stacks/
```
Comment (cnunciato) [inline content/docs/esc/integrations/infrastructure/pulumi-iac/pulumi-stacks.md:12]: Missing an alias.

File: content/docs/esc/integrations/dynamic-secrets/vault-secrets.md [hunk 11]
```diff
@@ -1,16 +1,15 @@
...
 menu:
   pulumiesc:
     identifier: vault-secrets
-    parent: esc-providers
-    weight: 9
+    parent: esc-dynamic-secrets
+    weight: 4
 aliases:
-- /docs/pulumi-cloud/esc/providers/vault-secrets/
+    - /docs/pulumi-cloud/esc/providers/vault-secrets/
```
Comment (cnunciato) [inline content/docs/esc/integrations/dynamic-secrets/vault-secrets.md:12]: Missing an alias.

File: content/docs/esc/integrations/dynamic-secrets/gcp-secrets.md [hunk 12]
```diff
@@ -1,16 +1,16 @@
...
 menu:
   pulumiesc:
     identifier: gcp-secrets
-    parent: esc-providers
-    weight: 6
+    parent: esc-dynamic-secrets
+    weight: 3
 aliases:
-- /docs/pulumi-cloud/esc/providers/gcp-secrets/
+    - /docs/pulumi-cloud/esc/providers/gcp-secrets/
```
Comment (cnunciato) [inline content/docs/esc/integrations/dynamic-secrets/gcp-secrets.md:13]: Missing an alias. (I'll stop commenting on these, but we should make sure every page that's been renamed has an alias that matches the path it's being moved from.)

File: content/docs/esc/integrations/infrastructure/_index.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
+---
+title: Infrastructure
```
Comment (cnunciato) [inline content/docs/esc/integrations/infrastructure/_index.md:2]: This heading feels a bit vague, and doesn't quite line up with the content underneath it (i.e., integrationg with Pulumi IaC, the Terraform CLI, and various aspects of CloudFlare and CloudFlare-related tooling).

File: content/docs/esc/other-integrations/_index.md
Comment (cnunciato) [inline content/docs/esc/other-integrations/_index.md:1]: Redirect?

File: content/docs/esc/providers/_index.md
Comment (cnunciato) [inline content/docs/esc/providers/_index.md:1]: Redirect?

File: docs.sln
Comment (cnunciato) [inline docs.sln:1]: Mind adding a line to the .gitignore for this one? It keeps sneaking in there somehow. 😄

**Review by cnunciato — CHANGES_REQUESTED** [review]

File: content/docs/esc/_index.md (hunk 3 above)
Comment (interurban) [reply content/docs/esc/_index.md:16]: This is the root docs URL path for ESC I would not suggest we change this, however I would suggest we create a "secrets-management" subpage in concepts to help with SEO.

Comment (interurban) [reply content/docs/esc/_index.md:16]: Issue filed here: https://github.com/pulumi/docs/issues/11524

File: content/docs/concepts/config.md [hunk 14]
```diff
@@ -616,7 +616,7 @@ Stack tags applied by Pulumi CLI are listed in the `Tags` section of the Overvie
 
 Often there is common configuration and secrets you do not want to duplicate in various stack configuration files. Pulumi ESC can help with that!
 
-Once you have an [environment](/docs/concepts/environments/) set up and you are [projecting pulumi configuration](/docs/concepts/environments/#projecting-pulumi-config), you can import that environment (or multiple environments) into your Pulumi stack.
+Once you have an [environment](/docs/esc/concepts/) set up and you are [projecting pulumi configuration](/docs/esc/environments/working-with-environments/#projecting-pulumi-config), you can import that environment (or multiple environments) into your Pulumi stack.
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/concepts/config.md:619]: added, thx!

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:1]: How about we  have this section expanded by default in the new nav.

File: content/docs/esc/concepts/_index.md [hunk 15]
```diff
@@ -1,17 +1,14 @@
...
-  concepts:
-    weight: 8
-search:
-  boost: true
-  keywords:
-    - environments
-    - secrets
-    - configuration
+  pulumiesc:
+    identifier: esc-concepts
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/concepts/_index.md:8]: There's a plan to add sub pages after launch.

File: content/docs/esc/development/_index.md [hunk 16]
```diff
@@ -0,0 +1,12 @@
+---
+title: Development
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/development/_index.md:2]: There's going to be new topics to this section added soon.

File: content/docs/esc/environments/_index.md [hunk 17]
```diff
@@ -0,0 +1,30 @@
+---
+title: Environments
+title_tag: Pulumi ESC Environments
+h1: Environments | Pulumi ESC
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/environments/_index.md:4]: thx, fixed.

File: content/docs/esc/environments/_index.md [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+
+Environments are accessible with the standalone [`esc` CLI](/docs/install/esc/), the [`pulumi` CLI](/docs/install/), the [Pulumi SDK](/docs/esc/languages-sdks/), and the [Pulumi Cloud console](/docs/esc/environments/working-with-environments#editing-an-environment-as-yaml-in-the-pulumi-cloud-console) and [REST API](/docs/pulumi-cloud/cloud-rest-api/#environments), and you can have as many environments as you need. Pulumi ESC is a service of Pulumi Cloud and is currently in public preview.
+
+## Key features
+
+Explore the following sections to learn more about the key features and benefits of Pulumi ESC:
+
+- [Working with environments](/docs/esc/environments/working-with-environments)
+- [Importing other environments](/docs/esc/environments/imports)
+- [Versioning](/docs/esc/environments/versioning)
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/environments/_index.md:24]: We'll have a new doc for this new page soon, but not yet.

File: content/docs/esc/environments/access-control.md [hunk 19]
```diff
@@ -0,0 +1,37 @@
+---
+title: Access control
+title_tag: Access control | Pulumi ESC
+h1: ESC access control
+meta_desc: Pulumi ESC provides granular access control to manage permissions with roles like reader, opener, and editor.
+menu:
+  pulumiesc:
+    identifier: esc-access-control
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/environments/access-control.md:8]: fixed

File: content/docs/esc/environments/working-with-environments.md (hunk 4 above)
Comment (interurban) [reply content/docs/esc/environments/working-with-environments.md:506]: nice catch, added.

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md (hunk 8 above)
Comment (interurban) [reply content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:14]: done

File: docs.sln
Comment (interurban) [reply docs.sln:1]: yup

File: content/docs/esc/other-integrations/_index.md
Comment (interurban) [reply content/docs/esc/other-integrations/_index.md:1]: yes its in the devtools page

File: content/docs/esc/providers/_index.md
Comment (interurban) [reply content/docs/esc/providers/_index.md:1]: yes added to integrations page

File: scripts/redirects/esc-cli-redirects.txt
Comment (interurban) [reply scripts/redirects/esc-cli-redirects.txt:1]: done, added

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md [hunk 20]
```diff
@@ -0,0 +1,98 @@
+---
+
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md]: fixed

File: content/blog/pulumi-release-notes-103/index.md [hunk 21]
```diff
@@ -38,19 +38,19 @@ As always, we have been actively rolling out new features in response to the inv
...
+We've rolled out the [Pulumi ESC Key-Value Table Editor](/blog/esc-key-value-table-editor-launch/), offering a novel, user-friendly interface for managing secrets and configurations. This addition complements the existing YAML-based editor, providing intuitive and familiar experiences for your workflows with secrets and configurations. Table Editor supports capabilities including CRUD operations for secrets and plaintext values, secret decryption, [environment imports](/docs/esc/environments/imports/#importing-other-environments), variable projections, and [interpolations](/docs/esc/environments/working-with-environments/#interpolating-values).
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/blog/pulumi-release-notes-103/index.md]: fixed

File: content/docs/esc/cli/command-line-completion.md [hunk 22]
```diff
@@ -1,12 +1,16 @@
...
 meta_image: /images/docs/meta-images/docs-meta.png
 menu:
-  esc_cli:
-    weight: 3
+  pulumiesc:
+    identifier: esc-command-line-completion
+    parent: esc-cli-overview
+    weight: 2
+aliases:
+    - docs/esc-cli/command-line-completion/
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/cli/command-line-completion.md]: fixed

**Comment by arunkumar611** [comment]
This part feels repetitive in both "Environments" and in "Working with Environments" 
image

I'd rather have this only in Environments page and have a table of contents on that page. 

Working with environments can start directly with creating a environment

**Comment by arunkumar611** [comment]
Update this to "Pulumi ESC Access Control". Do we need to state "Pulumi ESC" on the top of each page? Will it not be clear after the Nav update that the user is looking at ESC-only docs? 

image

**Comment by arunkumar611** [comment]
Add 1Password to the table of contents here:

image

**Comment by arunkumar611** [comment]
I'm good with everything else!

File: content/docs/esc/integrations/dynamic-secrets/gcp-secrets.md (hunk 12 above)
Comment (interurban) [reply content/docs/esc/integrations/dynamic-secrets/gcp-secrets.md:13]: fixed

File: content/docs/esc/development/_index.md [hunk 23]
```diff
@@ -0,0 +1,12 @@
...
+title_tag: Develop infrastructure solutions with Pulumi ESC
+h1: Develop with Pulumi ESC
+meta_desc: Pulumi ESC allows you to compose and manage hierarchical collections of configuration and secrets and consume them in various ways.
+menu:
+  pulumiesc:
+    identifier: esc-development
+    weight: 9
+---
+
+The [Pulumi ESC SDK](/docs/esc/development/languages-sdks/) provides a programmatic interface to manage environments, secrets, and configurations directly within your applications. Using the SDK, you can easily create, update, and delete environments, apply version tags, and manage the entire lifecycle of your infrastructure. It also simplifies the secure handling of secrets and configurations, allowing you to retrieve sensitive data like cloud credentials, connection strings, and feature flags at runtime without long-term storage.
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/development/_index.md]: fixed

File: content/docs/esc/environments/versioning.md (hunk 2 above)
Comment (interurban) [reply content/docs/esc/environments/versioning.md]: fixed

File: content/docs/esc/integrations/infrastructure/_index.md (hunk 13 above)
Comment (interurban) [reply content/docs/esc/integrations/infrastructure/_index.md:2]: Plan will be to leave this category title for now, assess after next week

File: content/docs/esc/environments/_index.md [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+Pulumi ESC (Environments, Secrets, and Configuration) lets you define collections of configuration settings and secrets called _environments_ and use them in any application or service. With Pulumi ESC, you can create environments, import configurations from a variety of sources, and manage secrets securely, ensuring your infrastructure is consistent and compliant.
+
+Environments are YAML documents composed of static key-value pairs, programmatic expressions, dynamically retrieved values from supported providers including all major clouds through OpenID Connect (OIDC), and other Pulumi ESC environments.
+
+Environments are accessible with the standalone [`esc` CLI](/docs/install/esc/), the [`pulumi` CLI](/docs/install/), the [Pulumi SDK](/docs/esc/languages-sdks/), and the [Pulumi Cloud console](/docs/esc/environments/working-with-environments#editing-an-environment-as-yaml-in-the-pulumi-cloud-console) and [REST API](/docs/pulumi-cloud/cloud-rest-api/#environments), and you can have as many environments as you need. Pulumi ESC is a service of Pulumi Cloud and is currently in public preview.
+
+## Key features
+
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/environments/_index.md]: Thx, removed this list of features and added an example.

File: content/docs/esc/environments/dynamic-environment-variables.md (hunk 1 above)
Comment (interurban) [reply content/docs/esc/environments/dynamic-environment-variables.md:15]: added to H1, removed h2

File: content/docs/esc/integrations/_index.md (hunk 5 above)
Comment (interurban) [reply content/docs/esc/integrations/_index.md]: fixed

File: content/docs/esc/integrations/_index.md (hunk 6 above)
Comment (interurban) [reply content/docs/esc/integrations/_index.md:25]: Understood, let's assess this into next week.

File: content/docs/esc/integrations/infrastructure/pulumi-iac/pulumi-stacks.md (hunk 10 above)
Comment (interurban) [reply content/docs/esc/integrations/infrastructure/pulumi-iac/pulumi-stacks.md]: fixed

File: content/docs/esc/integrations/dynamic-secrets/vault-secrets.md (hunk 11 above)
Comment (interurban) [reply content/docs/esc/integrations/dynamic-secrets/vault-secrets.md:12]: added

File: content/docs/esc/environments/dynamic-environment-variables.md [hunk 25]
```diff
@@ -0,0 +1,39 @@
+---
+title: Dynamic environment variables
```
[the remark this answers is not in the document]
Comment (interurban) [reply content/docs/esc/environments/dynamic-environment-variables.md:2]: added this as the H1

File: content/docs/esc/integrations/dynamic-secrets/_index.md (hunk 7 above)
Comment (interurban) [reply content/docs/esc/integrations/dynamic-secrets/_index.md:2]: Adjusted the H1 to be more descriptive

File: content/docs/esc/integrations/infrastructure/cloudflare.md (hunk 9 above)
Comment (interurban) [reply content/docs/esc/integrations/infrastructure/cloudflare.md:9]: fixed

**Review by sean1588 — APPROVED** [review]

**Pull request merged**